src/components/layout/page-header/PageHeader.tsx

```diff
@@ -64,104 +105,138 @@ export interface PageHeaderProps {
 
 /**
  * A flexible page header component that displays a title, optional metadata, description,
- * breadcrumbs, and actions. Can optionally include a divider, and <PageHeader.Navigation> elements
- * as children.
+ * breadcrumbs, actions and navigation.
  */
-export const PageHeaderRoot = forwardRef<HTMLDivElement, PageHeaderProps>((props, ref) => {
-  const { actions, breadcrumbs, description, hasDivider, metadata, navigation, title, ...rest } =
-    props;
+export const PageHeaderRoot = forwardRef<HTMLDivElement, PageHeaderProps>((props, forwardedRef) => {
+  const {
+    actions,
+    breadcrumbs,
+    description,
+    disableStickyActions = false,
+    hasDivider,
+    metadata,
+    navigation,
+    scrollBehavior = PageHeaderScrollBehavior.scroll,
+    title,
+    ...rest
+  } = props;
+
+  const [ref, setRef] = useForwardedRef(forwardedRef);
+  const { height = 0 } = useResizeObserver(ref);
+
+  // The actions are sticky when the PageHeader's behavior is `collapse`.
+  // `disableStickyActions` cancels that.
+  const stickyActions =
+    scrollBehavior === PageHeaderScrollBehavior.collapse && !disableStickyActions;
 
   return (
-    <StyledPageHeader ref={ref} {...rest}>
-      <StyledPageHeaderTop>
-        {breadcrumbs}
-
-        <StyledPageHeaderMain>
-          <StyledPageHeaderMainLeft>
-            {title}
-
-            {metadata}
-
-            {description}
-          </StyledPageHeaderMainLeft>
-
-          {actions && <StyledPageHeaderMainRight>{actions}</StyledPageHeaderMainRight>}
-        </StyledPageHeaderMain>
-      </StyledPageHeaderTop>
-
-      {(hasDivider || navigation) && (
-        <StyledPageHeaderBottom>
-          {navigation}
+    <StyledPageHeader
+      ref={setRef}
+      {...rest}
+      css={{
+        '--page-header-total-height': `${height}px`,
+        ...PageHeaderBehaviorStyles[scrollBehavior],
+      }}
+      hasFullWidthNav={!stickyActions}>
+      {breadcrumbs && <StyledPageHeaderBreadcrumbs>{breadcrumbs}</StyledPageHeaderBreadcrumbs>}
+
+      <StyledPageHeaderMain>
+        {title}
+
+        {metadata}
+
+        {description}
+      </StyledPageHeaderMain>
+
+      {actions && (
+        <StyledPageHeaderActions
+          style={
+            stickyActions
+              ? { top: cssVar('layout-page-header-actions-offset'), position: 'sticky' }
```

Comment:
This design token is only used in the context of a collapsed pageheader with sticky actions right? So maybe rename it to reflect that, right now it seems like a offset that's always applied to page header actions.